src/Core/Grid/Data/Factory/DoctrineGridDataFactory.php

```diff
@@ -93,9 +94,37 @@ public function getData(SearchCriteriaInterface $searchCriteria)
             'search_criteria' => $searchCriteria,
         ]);
 
-        $records = $searchQueryBuilder->execute()->fetchAll();
         $recordsTotal = (int) $countQueryBuilder->execute()->fetch(PDO::FETCH_COLUMN);
 
+        // If offset is out of range, we apply biggest valid offset.
+        if ($recordsTotal > 0 && $searchCriteria->getOffset() >= $recordsTotal) {
+            // Todo : those cases should be simplified with some refactoring cf. https://github.com/PrestaShop/PrestaShop/issues/29131
+            $searchLimit = !empty($searchCriteria->getLimit()) ? $searchCriteria->getLimit() : Filters::LIST_LIMIT;
```

Comment:
I think this would be same result
```suggestion
            $limit = (int) $searchCriteria->getLimit ?: Filters::LIST_LIMIT;
```

Reply:
but idk if we really need to restrict the ability to show full list? It would mean we remove the ability to have non-paginated lists :/

Reply:
:thinking: then do you recommend to fix the bug by adding something like this ?
```php
        if ($recordsTotal > 0 && $searchCriteria->getOffset() >= $recordsTotal) {

          // [...]

            if(empty($searchCriteria->getLimit()) {
              $inRangeOffset = $recordsTotal - 1;
            } else {
              $inRangeOffset = ($recordsTotal < $searchLimit) ? 0 : (floor($recordsTotal / $searchLimit) * $searchLimit - 1);
            }
```

Reply:
probably yes. Main idea is just not to provide fallback LIST_LIMIT value (so we make it possible to have unlimited lists).

Reply:
IMO it is not a good idea to have unlimited lists but as it is not the subject of this issue, you are right, I should keep original behavior a discuss about this somewhere else ;)